32-bit release build fails with warning on gcc 9

```
FAILED: core/CMakeFiles/dynamorio_static.dir/unix/loader.c.o 
/usr/bin/cc -DSTATIC_LIBRARY -I/home/bruening/dr/git/src/core/drlibc -I/home/bruening/dr/git/src/core/arch/x86 -I/home/bruening/dr/git/src/core/unix -I/home/bruening/dr/git/src/core/arch -I/home/bruening/dr/git/src/core/lib -I. -Iinclude/annotations -m32  -fno-strict-aliasing -fno-stack-protector -mpreferred-stack-boundary=2 -fvisibility=internal  -std=gnu99 -fno-unwind-tables -O3 -g3 -Wall -Werror -Wwrite-strings -Wno-unused-but-set-variable -Wno-stringop-truncation -Wno-format-truncation -Wno-stringop-overflow     -fPIC -MD -MT core/CMakeFiles/dynamorio_static.dir/unix/loader.c.o -MF core/CMakeFiles/dynamorio_static.dir/unix/loader.c.o.d -o core/CMakeFiles/dynamorio_static.dir/unix/loader.c.o   -c /home/bruening/dr/git/src/core/unix/loader.c
/home/bruening/dr/git/src/core/unix/loader.c: In function ‘privload_call_entry’:
/home/bruening/dr/git/src/core/unix/loader.c:990:5: error: listing the stack pointer register ‘esp’ in a clobber list is deprecated [-Werror=deprecated]
  990 |     __asm__ __volatile__("mov %%esp, %%edi\n"       /* Save the pre-alignment sp. */
      |     ^~~~~~~
/home/bruening/dr/git/src/core/unix/loader.c:990:5: note: the value of the stack pointer after an ‘asm’ statement must be the same as it was before the statement
/home/bruening/dr/git/src/core/unix/loader.c:990:5: error: listing the stack pointer register ‘esp’ in a clobber list is deprecated [-Werror=deprecated]
  990 |     __asm__ __volatile__("mov %%esp, %%edi\n"       /* Save the pre-alignment sp. */
      |     ^~~~~~~
/home/bruening/dr/git/src/core/unix/loader.c:990:5: note: the value of the stack pointer after an ‘asm’ statement must be the same as it was before the statement
/home/bruening/dr/git/src/core/unix/loader.c:990:5: error: listing the stack pointer register ‘esp’ in a clobber list is deprecated [-Werror=deprecated]
  990 |     __asm__ __volatile__("mov %%esp, %%edi\n"       /* Save the pre-alignment sp. */
      |     ^~~~~~~
/home/bruening/dr/git/src/core/unix/loader.c:990:5: note: the value of the stack pointer after an ‘asm’ statement must be the same as it was before the statement
/home/bruening/dr/git/src/core/unix/loader.c:990:5: error: listing the stack pointer register ‘esp’ in a clobber list is deprecated [-Werror=deprecated]
  990 |     __asm__ __volatile__("mov %%esp, %%edi\n"       /* Save the pre-alignment sp. */
      |     ^~~~~~~
/home/bruening/dr/git/src/core/unix/loader.c:990:5: note: the value of the stack pointer after an ‘asm’ statement must be the same as it was before the statement
cc1: all warnings being treated as errors
```

--- a/core/unix/loader.c
+++ b/core/unix/loader.c
@@ -998,7 +998,10 @@ privload_call_lib_func(fp_t func)
                          :
                          : [env] "g"(our_environ), [argv] "g"(&dummy_argv[0]),
                            [callee] "g"(func)
-                         : "edi", "esp", "memory");
+                         /* We do *not* list "esp" because doing so is disallowed
+                          * (i#4086).  We do restore esp so we're fine.
+                          */
+                         : "edi", "memory");
 #endif
 }
 

